src/main/java/seedu/address/ui/CalendarView.java

```diff
@@ -0,0 +1,330 @@
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import javafx.scene.text.Font;
+import javafx.scene.text.FontWeight;
+
+/**
+ * UI component that is displayed.
+ */
+public class CalendarView extends UiPart<Region> {
+
+    private static final String FXML = "CalendarView.fxml";
+    private static final String[] MONTHS = {"January", "February", "March", "April", "May", "June", "July", "August",
+        "September", "October", "November", "December"};
+    private static final int[] DAYS_IN_MONTH = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+    private int[] simulateGridPane = new int [42];
+    private int day;
+    private int month;
+    private int year;
+    private YearMonth yearMonth;
+    private LocalDate pivotDate;
+    private LocalDate firstDayOfTheMonth;
+    private int prevMonthBalance;
+    private int nextMonthBalance;
+    private int thisMonthBalance;
+
+    private Image leftArrow = new Image(this.getClass().getResourceAsStream("/images/leftButton.png"),
+             20, 15, true, true);
+    private Image rightArrow = new Image(this.getClass().getResourceAsStream("/images/rightButton.png"),
+             20, 15, true, true);
+
+    @FXML
+    private Label monthYearLabel;
+
+    @FXML
+    private GridPane dateGridPane;
+
+    @FXML
+    private GridPane monthYearGridPane;
+
+    @FXML
+    private GridPane weekDayGridPane;
+
+    @FXML
+    private Button leftButton;
+
+    @FXML
+    private Button rightButton;
+
+    public CalendarView() {
+        super(FXML);
+        setUpButton();
+        LocalDate now = LocalDate.now();
+        this.pivotDate = now;
+        this.day = now.getDayOfMonth();
+        this.month = now.getMonthValue();
+        this.year = now.getYear();
+        this.yearMonth = YearMonth.of(this.year, this.month);
+        this.firstDayOfTheMonth = yearMonth.atDay(1);
+        setMonthYearLabel();
+        generateCalender();
+    }
+
+    public CalendarView(LocalDate date) {
+        super(FXML);
+        setUpButton();
+        this.pivotDate = date;
+        this.day = date.getDayOfMonth();
+        this.month = date.getMonthValue();
+        this.year = date.getYear();
+        this.yearMonth = YearMonth.of(this.year, this.month);
+        this.firstDayOfTheMonth = yearMonth.atDay(1);
+        setMonthYearLabel();
+        generateCalender();
+    }
+
+    /**
+     * Set up the left and right arrow button.
+     */
+    private void setUpButton() {
+        ImageView leftButtonView = new ImageView(this.leftArrow);
+        ImageView rightButtonView = new ImageView(this.rightArrow);
+        leftButton.setGraphic(leftButtonView);
+        rightButton.setGraphic(rightButtonView);
+    }
+
+    /**
+     * Check if a given year is a leap year.
+     * @param year input year
+     * @return boolean, true to denote leap year else false.
+     */
+    private boolean checkLeapYear(int year) {
+        if (year % 4 == 0) {
+            return true;
+        }
```

Comment:
There are more conditions for leap year:
- divisible by 100 is not a leap year
- but divisible by 400 is